week04/Detect Squares.cpp

```diff
@@ -0,0 +1,31 @@
+class DetectSquares {
+private:
+    int counts[1001][1001]={};
```

Comment:
This is not acceptable. You are using too much memory for each test case. Please don't assume memory is unlimited and you can use whenever wherever however you want.